packages/opentelemetry-cloud-trace-exporter/src/transform.ts

```diff
@@ -173,14 +174,26 @@ function mergeAttributes(...attributeList: Attributes[]): Attributes {
 
 function transformResourceToAttributes(
   resource: Resource,
-  projectId: string
+  projectId: string,
+  resourceFilter?: RegExp
 ): Attributes {
   const monitoredResource = mapOtelResourceToMonitoredResource(
     resource,
     projectId
   );
   const attributes: ot.SpanAttributes = {};
 
+  if (resourceFilter) {
+    const filteredResourceAttributes: [string, ot.SpanAttributeValue][] =
+      Object.keys(resource.attributes)
+        .filter(key => resourceFilter.test(key))
+        .map(key => [key, resource.attributes[key]]);
+
+    for (const [key, value] of filteredResourceAttributes) {
+      attributes[key] = value;
+    }
```

Comment:
Something like this should work and can remove the const altogether.

```suggestion
        .forEach(key => { attributes[key] = resource.attributes[key]; });
```